src/main/flight/imu.c

```diff
@@ -219,7 +219,15 @@ static void imuMahonyAHRSupdate(float dt, float gx, float gy, float gz,
             courseOverGround += (2.0f * M_PIf);
         }
 
-        const float ez_ef = (- sin_approx(courseOverGround) * rMat[0][0] - cos_approx(courseOverGround) * rMat[1][0]);
+        float ez_ef = (- sin_approx(courseOverGround) * rMat[0][0] - cos_approx(courseOverGround) * rMat[1][0]);
+
+#ifdef USE_GPS_RESCUE
```

Comment:
(good enough for testing)
RESCUE related code should not be here. Replace `useCog` with gain and move this code to `imuMahonyAHRSupdate` call place.

Reply:
(just tested, handled 'normal' rescues in still air with moderate IMU disorientation very well)

I'll try to do as you say, sorry for not getting it right, but I am not much good as a coder.

Reply:
It's just cosmetics - keep IMU code as encapsulated as possible

Reply:
I looked at trying to use your suggestion to make `useCOG` as a float, but I see that `useCOG` is a logical operator in several places.
In particular I've never quite followed the logic of these lines:
https://github.com/betaflight/betaflight/pull/12738/files#diff-de6b2b61376cbb30512686ebb362260facb69cf9a4281ebdcd59af61c0306f7aL500-L506
I wouldn't want to mess this up.

Reply:
What about `useCOG ? imuCogGain : 0.0f`. And in Mahony update, rename parameter to `float cogGain`, but still test for zero (to skip calculation and avoid unlikely case when  CoG is NaN)

Reply:
If it's OK I would like to look at this code idea in a separate PR.  
I say that because we have this arbitrary 2.0m/s on/off behaviour of the mahony update.  I am wondering if this should be re-considered.   Perhaps it could be a gain factor rather than an on/off method.
I would like to look at refactoring the useCog and consider related issues later because this PR is functionally solid now, and it may take a while to do the useCog refactoring.

Reply:
> Perhaps it could be a gain factor rather than an on/off method
You still want zero gain (no CoG) on low speed. But you can ramp gain when minimum speed is exceeded.

Modifying CoG gain inside inner IMU code is ugly. It should be refactored ASAP. Only few lines of code is needed.

Reply:
@ledvinap I converted `useCOG` to a float `cogYawGain` in commit [9af3cb8ae](https://github.com/betaflight/betaflight/pull/12738/commits/9af3cb8ae5aa5747cd7ee95d453a37b31a0021d2), and it seems to work very well.  Thanks for encouraging me to do this.
Also I increased the rate of acquisition of gain by a factor of 2x, and tested it extensively.  I didn't have a lot of wind, but did test with 180 degree IMU errors and a wide range of velocities, including >120kph away from home, and it worked better than before.  The IMU correction was quicker, obviously, leading to less overshoot and tighter radius corrections.  I was concerned that it might rotate excessively with severe errors but I think the gain was at the limit of 6x on previous tests anyway, and in any case there were some rescues where the IMU gain resulted in an overcorrection, but this was quickly resolved.  

The worst performing situation now is where there is 180 degree IMU disorientation and the quad is 'dead still' in the air at the time of the rescue.  This is below the 2m/s threshold for IMU adjustment.  It must get more than 2m/s before the IMU will even consider the groundspeed.  With no groundspeed to start with, quad must pitch forward to gain velocity.  Since there is little pitch angle attenuation, to start with, it starts to gain quite a bit of speed over time.  With the lag in GPS data, it takes some time before the velocity change results in increase in IMU gain, and by then the velocity is appreciable.  The turn is initially a bit slower than you'd like, and it can seem like it just won't turn at all.  Hence the overshoot distance traveled is further than you'd like, maybe 50m, but once the groundspeed has built up, and is properly detected, the correction becomes fast.  This results in some over-correction, but that is quite good, since it brings you back towards the intended path).  

Hence a user in this circumstance needs about 50m of clear air around them to cover their worst possible outcome.  This is probably a good general recommendation for this code.  At high speed the relative problem seems less but the absolute drift is quite large, just the pre-existing drift speed during the climb results in a lot of travel in those situations.  

In all other cases, having some pre-existing drift when the IMU is 180 degrees wrong seems to make things easier and quicker now; if the drift is away from home, there is already some velocity away, leading to a very quick fix.  If the drift is towards home, but you got the IMU 180 wrong, it will still rotate and fly away from home, but the net effect isn't as bad as a standing start, because we cross the 2m/s 'dead spot' with reasonable velocity already acquired.  Sideways drifts and IMU errors at less than 180 degree all have some groundspeed which helps the IMU to adjust more quickly.

Overall this is pretty good I think.